tests/functional/watcher_controller_test.go

```diff
@@ -0,0 +1,127 @@
+package functional
+
+import (
+	. "github.com/onsi/ginkgo/v2" //revive:disable:dot-imports
+	. "github.com/onsi/gomega"    //revive:disable:dot-imports
+
+	//revive:disable-next-line:dot-imports
+	condition "github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	. "github.com/openstack-k8s-operators/lib-common/modules/common/test/helpers"
+	watcherv1beta1 "github.com/openstack-k8s-operators/watcher-operator/api/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	MinimalWatcherSpec = map[string]interface{}{
+		"databaseInstance": "openstack",
+	}
```

Comment:
note: eventually the minimal empty spec will likely be just the the kind: "watcher" and an empty doc as this will be added by the defaulting webhook

but this is perfectly fine for now.